service/security/test/src/test/resources/features/RoleServiceUnitTests.feature

```diff
@@ -0,0 +1,286 @@
+     Given I prepare a role creator with name "roleName" and description "roleDescription"
+     When I create a new role entity from the existing creator
+     And I update the role name to "roleName2"
+     Then I find a role with name "roleName2"
+     And No exception was thrown
+
+   Scenario: Changing role description to a valid one
+     Create a role with a valid name and valid description (e.g "roleDescription").
+     After that try to edit role description to e.g "roleDescription123.
+     Kapua should not return any errors.
+
+     Given I prepare a role creator with name "roleName" and description "roleDescription"
+     When I create a new role entity from the existing creator
+     And I update the role description to "roleDescription123"
+     Then I search for a role with description "roleDescription"
+     And I find no roles
+
+   Scenario: Changing name of a nonexisting role
+     Create a role with a valid name (e.g "roleName") and with a valid description (e.g "roleDescription").
+     After that delete it and try to update it.
+     Kapua should return an error.
+     
+     Given I prepare a role creator with name "roleName" and description "roleDescription"
+     When I create a new role entity from the existing creator
+     Then I delete the role with name "roleName"
+     But I expect the exception "KapuaEntityNotFoundException" with the text "*"
+     And I update the last created role name to "roleName2"
+     Then An exception was thrown
+
+   Scenario: Changing description of a nonexisting role
+     Create a role with a valid name (e.g "roleName") and with a valid description (e.g "roleDescription").
+     After that delete it and try to update role description.
+     Kapua should return an error.
+
+     Given I prepare a role creator with name "roleName" and description "roleDescription"
+     When I create a new role entity from the existing creator
+     Then I delete the role with name "roleName"
+     But I expect the exception "KapuaEntityNotFoundException" with the text "*"
+     And I update the role description to "roleDescription123"
+     And An exception was thrown
+
+   Scenario: Changing role name to null
+     Create a role with a valid name (e.g "roleName") and with a valid description (e.g "roleDescription").
+     After that tr to update role name to "null".
+     Kapua should return an error.
+
+     Given I prepare a role creator with name "roleName" and description "roleDescription"
+     When I create a new role entity from the existing creator
+     But I expect the exception "KapuaIllegalNullArgumentException" with the text "An illegal null value was provided for the argument role.name."
+     Then I update the role name to ""
+     And An exception was thrown
+
+   Scenario: Changing role name to contain special character
+     Create a role with name e.g "roleName" and with description e.g "roleDescription",
+     then try to update role name with special characters e.g "roleName$%^".
```

Comment:
Change description line to: then try to update role name with special characters( e.g "roleName%")."